epsagon/events/dbapi.py

```diff
@@ -67,30 +67,39 @@ def __init__(
         super(DBAPIEvent, self).__init__(start_time)
 
         self.event_id = 'dbapi-{}'.format(str(uuid4()))
-        dsn = parse_dsn(connection.dsn)
-        self.resource['name'] = dsn['dbname']
 
-        query = cursor.query
+        if hasattr(connection, 'dsn'):
+            dsn = parse_dsn(connection.dsn)
+            db_name = dsn['dbname']
+            host = dsn.get('host', 'local')
+            query = cursor.query
+        else:
+            query = _args[0]
+            host = connection.extract_hostname
+            db_name = connection.extract_dbname
+
+        self.resource['name'] = db_name
```

Comment:
What happen if for example we get an exception here? I don't see any catch around here(?)

Reply:
It's safe. The general wrapper is catching any exception, this will be reported as an instrumentation exception.